test/support/factories/detour_factory.ex

```diff
@@ -74,12 +75,23 @@ defmodule Skate.DetourFactory do
         put_in(state["value"], %{"Detour Drawing" => "Past"})
       end
 
-      def with_route_name(%Skate.Detours.Db.Detour{} = detour, route_name) do
-        %{detour | state: with_route_name(detour.state, route_name)}
+      def with_route(%Skate.Detours.Db.Detour{} = detour, route) do
+        %{detour | state: with_route(detour.state, route)}
       end
 
-      def with_route_name(%{"context" => %{"route" => %{"name" => _}}} = state, route_name) do
-        put_in(state["context"]["route"]["name"], route_name)
+      def with_route(
+            %{
+              "context" => %{
+                "route" => %{"name" => _, "id" => _, "directionNames" => direction_names}
+              }
+            } = state,
+            route
+          ) do
+        put_in(state["context"]["route"], %{
+          "name" => route,
+          "id" => route,
+          "directionNames" => direction_names
+        })
```

Comment:
I think we should have separate functions for `with_route` and `with_route_id` because in this form, it's not possible to have name be SL3 and id be 743, which the bug which introduced this change is based on them differing.

Reply:
Oh my goodness OF COURSE! Ok and I need to add a test for that specifically. Smh for missing that, but thank you

Reply:
Well... maybe a test isn't the most useful in this case. But properly setting id / name is the right change to make at least